### PR TITLE
Feature/ts rewrite

### PR DESCRIPTION
**Pull request by michielvandergeest**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -15,19 +15,802 @@
...
+
+  interface Transition {
+    /**
+     * Name of the prop to transition (i.e. 'x', 'y', 'alpha', 'color')
+     */
+    prop: string,
+    /**
+     * Value the prop should transition to (i.e. 0, 100, '#223388')
+     */
+    value: any,
```
Comment (wouterlucas) [inline index.d.ts:372]: are these typeable as well ? e.g. number | string | boolean | CustomType?

File: index.d.ts [hunk 2]
```diff
@@ -15,19 +15,802 @@
...
+
+  interface Before {
+    /**
+    * Name of the prop to set before the transition starts
+    */
+    prop: string,
+    /**
+     * Value the prop to set before the transition starts
+     */
+    value: any,
```
Comment (wouterlucas) [inline index.d.ts:395]: same here, can we add a type here?

File: index.d.ts [hunk 3]
```diff
@@ -15,19 +15,802 @@
...
+    /**
+     * Message to be announced
+     */
+    message: String,
+    /**
+     * Politeness level
+     *
+     * Defaults to 'off'
+     */
+    politeness?: 'off' | 'polite' | 'assertive'
```
Comment (wouterlucas) [inline index.d.ts:425]: what does this do?

File: index.d.ts [hunk 4]
```diff
@@ -15,19 +15,802 @@
...
+    */
+    type: 'msdf' | 'sdf',
+  } & (SdfFontWithFile | SdfFontWithPngJson)
+
+  type Font = WebFont | SdfFont
+
+  type ShaderEffect = {
+    name: string,
+    type: RendererShaderEffect
+  }
```
Comment (wouterlucas) [inline index.d.ts:538]: I guess this works for now - but we should think about making this easier when the new shader / effect refactor from Jeff comes in :)

**Review by wouterlucas — APPROVED** [review]
LGTM!

**Review by suresh-gangumalla — APPROVED** [review]
LGTM

File: index.d.ts (hunk 3 above)
Comment (michielvandergeest) [reply index.d.ts:425]: it's a property used in text to speech to indicate critical notifications that require immediate attention

File: index.d.ts (hunk 4 above)
Comment (michielvandergeest) [reply index.d.ts:538]: yes, this is temporary for now, and up for general refactor

**Pull request merged**